matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/poll/DefaultPollHistoryService.kt

```diff
@@ -0,0 +1,155 @@
+import timber.log.Timber
+
+private const val LOADING_PERIOD_IN_DAYS = 30
+private const val EVENTS_PAGE_SIZE = 250
+
+internal class DefaultPollHistoryService @AssistedInject constructor(
+        @Assisted private val roomId: String,
+        @Assisted private val timelineService: TimelineService,
+        @SessionDatabase private val monarchy: Monarchy,
+        private val clock: Clock,
+        private val loadMorePollsTask: LoadMorePollsTask,
+        private val getLoadedPollsStatusTask: GetLoadedPollsStatusTask,
+        private val syncPollsTask: SyncPollsTask,
+        private val timelineEventMapper: TimelineEventMapper,
+) : PollHistoryService {
+
+    @AssistedFactory
+    interface Factory {
+        fun create(roomId: String, timelineService: TimelineService): DefaultPollHistoryService
+    }
+
+    override val loadingPeriodInDays: Int
+        get() = LOADING_PERIOD_IN_DAYS
+
+    private val timeline by lazy {
+        val settings = TimelineSettings(
+                initialSize = EVENTS_PAGE_SIZE,
+                buildReadReceipts = false,
+                rootThreadEventId = null,
+                useLiveSenderInfo = false,
+        )
+        timelineService.createTimeline(eventId = null, settings = settings).also { it.start() }
+    }
+    private val timelineMutex = Mutex()
+
+    override fun dispose() {
+        timeline.dispose()
+    }
+
+    override suspend fun loadMore(): LoadedPollsStatus {
+        return timelineMutex.withLock {
+            val params = LoadMorePollsTask.Params(
+                    timeline = timeline,
+                    roomId = roomId,
+                    currentTimestampMs = clock.epochMillis(),
+                    loadingPeriodInDays = loadingPeriodInDays,
+                    eventsPageSize = EVENTS_PAGE_SIZE,
+            )
+            loadMorePollsTask.execute(params)
+        }
+    }
+
+    override suspend fun getLoadedPollsStatus(): LoadedPollsStatus {
+        val params = GetLoadedPollsStatusTask.Params(
+                roomId = roomId,
+                currentTimestampMs = clock.epochMillis(),
+        )
+        return getLoadedPollsStatusTask.execute(params)
+    }
+
+    override suspend fun syncPolls() {
+        timelineMutex.withLock {
+            val params = SyncPollsTask.Params(
+                    timeline = timeline,
+                    roomId = roomId,
+                    currentTimestampMs = clock.epochMillis(),
+                    eventsPageSize = EVENTS_PAGE_SIZE,
+            )
+            syncPollsTask.execute(params)
+        }
+    }
+
+    override fun getPollEvents(): LiveData<List<TimelineEvent>> {
+        val pollHistoryStatusLiveData = getPollHistoryStatus()
+
+        return Transformations.switchMap(pollHistoryStatusLiveData) { results ->
+            val oldestTimestamp = results.firstOrNull()?.oldestTimestampTargetReachedMs ?: clock.epochMillis()
+            Timber.d("oldestTimestamp=$oldestTimestamp")
```

Comment:
Should this log have been removed?

Reply:
Yes thanks for catching it! Removed in [fc26d61](https://github.com/vector-im/element-android/pull/8003/commits/fc26d61305849a33ee8f9a40e758d6edee06b02e)